src/Provider/AbstractProvider.php

```diff
@@ -21,11 +24,22 @@
  */
 abstract class AbstractProvider implements ProviderInterface
 {
-    protected $httpAdapter;
+    /**
+     * @var HttpClient
+     */
+    private $httpClient;
+
+    /**
+     * @var RequestFactory
+     */
+    private $httpMessageFactory;
 
-    public function __construct(HttpAdapterInterface $httpAdapter)
+    /**
+     * @param HttpClient $httpClient
+     */
+    public function __construct(HttpClient $httpClient = null)
```

Comment:
Sure, we could but I do not think many people will use that. I do not people will care what PSR-7 implementation that are used by Swap internally. And if they do they have the setter. 

The discovery 1.0 is very lightweight now compared to when I first made this PR. There is no longer an issue for people to use the discovery layer.